src/iter/mod.rs

```diff
@@ -649,8 +649,9 @@ pub trait ParallelIterator: Sized {
     /// Partitions the items of a parallel iterator into a pair of arbitrary
     /// `ParallelExtend` containers.  Items for which the `predicate` returns
     /// true go into the first container, and the rest go into the second.
-    fn partition<C, P>(self, predicate: P) -> (C, C)
-        where C: Default + ParallelExtend<Self::Item>,
+    fn partition<A, B, P>(self, predicate: P) -> (A, B)
+        where A: Default + ParallelExtend<Self::Item>,
+              B: Default + ParallelExtend<Self::Item>,
```

Comment:
Yay! My only hesitation is that it means that transition to rayon *may* require more type annotations. e.g., with seq iterators, people can do:

```rust
let (x, y): (Vec<_>, _) = iter.partition();
```

maybe we should add another method for this case? (We could then propose the same method for seq iterators, I suppose). But I don't know what would be a good name.